buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -0,0 +1,527 @@
+            task.dependsOn(sourcesJarProvider);
+        });
+    }
+
+    private static TaskProvider<ApplyAccessTransformer> configureAccessTransformer(
+            Project project,
+            NeoDevConfigurations configurations,
+            TaskProvider<CreateMinecraftArtifacts> createSourceArtifacts,
+            Provider<Directory> neoDevBuildDir,
+            File atFile) {
+
+        return project.getTasks().register("applyAccessTransformer", ApplyAccessTransformer.class, task -> {
+            task.setGroup(INTERNAL_GROUP);
+            task.classpath(configurations.getExecutableTool(Tools.JST));
+            task.getInputJar().set(createSourceArtifacts.flatMap(CreateMinecraftArtifacts::getSourcesArtifact));
+            task.getAccessTransformer().set(atFile);
+            task.getOutputJar().set(neoDevBuildDir.map(dir -> dir.file("artifacts/access-transformed-sources.jar")));
+            task.getLibraries().from(configurations.neoFormClasspath);
+            task.getLibrariesFile().set(neoDevBuildDir.map(dir -> dir.file("minecraft-libraries-for-jst.txt")));
+        });
+    }
+
+    private static BinaryPatchOutputs configureBinaryPatchCreation(Project project,
+                                                                   NeoDevConfigurations configurations,
+                                                                   TaskProvider<CreateCleanArtifacts> createCleanArtifacts,
+                                                                   Provider<Directory> neoDevBuildDir,
+                                                                   File sourcesPatchesFolder) {
+        var tasks = project.getTasks();
+
+        var artConfig = configurations.getExecutableTool(Tools.AUTO_RENAMING_TOOL);
+        var remapClientJar = tasks.register("remapClientJar", RemapJar.class, task -> {
+            task.setDescription("Creates a Minecraft client jar with the official mappings applied. Used as the base for generating binary patches for the client.");
+            task.getObfSlimJar().set(createCleanArtifacts.flatMap(CreateCleanArtifacts::getCleanClientJar));
+            task.getMojmapJar().set(neoDevBuildDir.map(dir -> dir.file("remapped-client.jar")));
+        });
+        var remapServerJar = tasks.register("remapServerJar", RemapJar.class, task -> {
+            task.setDescription("Creates a Minecraft dedicated server jar with the official mappings applied. Used as the base for generating binary patches for the client.");
+            task.getObfSlimJar().set(createCleanArtifacts.flatMap(CreateCleanArtifacts::getCleanServerJar));
+            task.getMojmapJar().set(neoDevBuildDir.map(dir -> dir.file("remapped-server.jar")));
+        });
+        for (var remapTask : List.of(remapClientJar, remapServerJar)) {
+            remapTask.configure(task -> {
+                task.setGroup(INTERNAL_GROUP);
+                task.classpath(artConfig);
+                task.getMainClass().set("net.neoforged.art.Main");
```

Comment:
The hardcoding of the class name bothers me, I would read the `Main-Class` attribute from the tool jar when it uses full classpath. In any case, this should be moved to the `Tools` at least.

Reply:
Turns out we're currently using the fatjar, so this line can be removed.